pkg/addonmanager/controllers/addoninstall/controller.go

```diff
@@ -98,6 +108,27 @@ func (c *addonInstallController) sync(ctx context.Context, syncCtx factory.SyncC
 			continue
 		}
 
+		cma, err := c.clusterManagementAddonLister.Get(addonName)
```

Comment:
this will be called per mca/cluster event. It might bring a lot of conflict. How about we put this in a controller that reconcile clustermanagementaddon only?

Reply:
Yes, that would be a better choice. The current code will have conflicts when the first time adding the annotation, then won't trigger patch action later.

Reply:
we have another existing controller that is reconciling the clustermanageraddon